keras_nlp/layers/transformer_encoder_test.py

```diff
@@ -0,0 +1,144 @@
+
+import tensorflow as tf
+
+from keras_nlp.layers import transformer_encoder
+
+
+class TransformerEncoderTest(TestCase):
+    def test_valid_call(self):
+        encoder = transformer_encoder.TransformerEncoder(
+            feedforward_dim=4,
+            num_heads=2,
+        )
+        model = tf.keras.Sequential(
+            [
+                tf.keras.Input(shape=(4, 6)),
+                encoder,
+            ]
+        )
+        input = tf.random.uniform(shape=[2, 4, 6])
+        model(input)
+
+    def test_valid_call_with_mask(self):
+        encoder = transformer_encoder.TransformerEncoder(
+            feedforward_dim=4,
+            num_heads=2,
+        )
+        encoder.build([2, 4, 6])
+        input = tf.random.uniform(shape=[2, 4, 6])
+        mask = input[:, :, 0] < 0.5
+        encoder(input, mask)
+
+    def test_get_config_and_from_config(self):
+        encoder = transformer_encoder.TransformerEncoder(
+            feedforward_dim=4,
+            num_heads=2,
+        )
+        config = encoder.get_config()
+        expected_config_subset = {
+            "feedforward_dim": 4,
+            "num_heads": 2,
+            "dropout": 0,
+            "activation": "relu",
+            "layer_norm_epsilon": 1e-05,
+        }
+        self.assertEqual(config, {**config, **expected_config_subset})
+
+        restored_encoder = transformer_encoder.TransformerEncoder.from_config(
+            config,
+        )
+        assert restored_encoder._feedforward_dim == 4
+        assert restored_encoder._num_heads == 2
+
+    def test_one_training_step_of_transformer_encoder(self):
+        encoder = transformer_encoder.TransformerEncoder(
+            feedforward_dim=4,
+            num_heads=2,
+        )
+        inputs = tf.keras.Input(shape=(4, 6))
+        x = encoder(inputs)
+        x = tf.keras.layers.Dense(1, activation="sigmoid")(x)
+        model = tf.keras.Model(inputs=inputs, outputs=x)
+
+        data = tf.random.uniform(shape=[2, 4, 6])
+        label = tf.cast(data[:, :, 0] >= 0.5, dtype=tf.int32)
+
+        loss_fn = tf.keras.losses.BinaryCrossentropy(from_logits=False)
+        optimizer = tf.keras.optimizers.Adam()
+        with tf.GradientTape() as tape:
+            pred = model(data)
+            loss = loss_fn(label, pred)
+        grad = tape.gradient(loss, model.trainable_variables)
+        self.assertTrue(len(grad) > 1)
+        optimizer.apply_gradients(zip(grad, model.trainable_variables))
+
+    def test_checkpointing_transformer_encoder(self):
+        encoder1 = transformer_encoder.TransformerEncoder(
+            feedforward_dim=4,
+            num_heads=2,
+        )
+
+        encoder2 = transformer_encoder.TransformerEncoder(
+            feedforward_dim=4,
+            num_heads=2,
+        )
+        data = tf.random.uniform(shape=[2, 4, 6])
+        encoder1(data)
+        encoder2(data)
+        # The weights of encoder1 and encoder2 are different.
+        self.assertFalse(
```

Comment:
We have a few assertTrue assertFalse that seem to have leaked in. Will this even work with pytest?

Reply:
pytest has support for unittest, and running these tests passed, so we are good here?